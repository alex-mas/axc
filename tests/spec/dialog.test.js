import React from 'react'
import { shallow, render, mount } from 'enzyme'
import Dialog from '../../dist/layout/dialog'


test('Dialog should crash if not passed any option', () => {
    const onClose = jest.fn();
    expect(()=>{
        let wrapper = shallow(
            <Dialog
                isOpen={false}
                onClose={onClose}
            >
                <div>
                    this should not render at all
                </div>
            </Dialog>
        );
    }).toThrow();
});


test('Dialog should not render anything if isOpen is false', () => {
    const onClose = jest.fn();
    let wrapper = shallow(
        <Dialog
            isOpen={false}
            onClose={onClose}
            options={["first optipon","second option"]}
        >
            <div id="dialog__content">
                this should not render at all
            </div>
        </Dialog>
    );
    expect(wrapper.find).toMatchSnapshot();
    expect(wrapper.find("#dialog__content").exists()).toEqual(false);
});