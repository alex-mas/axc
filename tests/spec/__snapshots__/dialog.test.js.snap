// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog should render the dialog properly if isOpen equals to true 1`] = `[Function]`;

exports[`Dialog shouldn't render anything if isOpen is false 1`] = `[Function]`;
